ultimarc/devices/mini_pac.py

```diff
@@ -119,22 +120,66 @@ def _create_message_(self, config_file, cur_device_config=None):
             data.header.byte_3 = 0x0f
             data.header.byte_4 = header.asByte
 
-            # TODO: Allow macros to be optional and figure out macro naming functionality.
-            #  Does it have to be like the old way?
+            # key: Macro name value: macro value (e0 - fe)
+            macro_dict = {}
+            try:
+                # Macros
+                # Each macro starts with control character e0 - fe.  Total of 30 macros possible
+                # overall total macro characters is 85
+                max_size = 85
+                max_macro = 30
+                cur_size_count = 0
+                cur_macro_count = 0
+                cur_macro = 0xe0
+                cur_position = 166
+
+                for macro in config.macros:
+                    if len(macro.action) > 0:
```

Comment:
I think you could just put an if statement before the 'for' statement to check the number of macros.  IE: 'if len(config.macros) > max_macros:', instead of in the 'for' loop.

Reply:
Moved the max macro entries test out of the for loop.  Added a test for max entries(30) and max actions (85).  Also fixed a check above for a valid configuration object.